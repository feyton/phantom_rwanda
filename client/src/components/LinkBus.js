import React from 'react';
import { Link } from 'react-router-dom';

export const LinkBus = ({ row }) => {
  const { bus } = row.original;
  if (bus)
    return (
      <Link className="text-primary" to="#buses">
        {bus.plateNumber}
      </Link>
    );

  return 'None';
};
