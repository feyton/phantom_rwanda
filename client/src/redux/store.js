import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { busMovementReducer } from './reducers/busMovementReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { simulateReducer } from './reducers/simulateReducer.js';
import { socketReducer } from './reducers/socketReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer,
    socket: socketReducer,
    simulate: simulateReducer,
    movement: busMovementReducer
  }
});
