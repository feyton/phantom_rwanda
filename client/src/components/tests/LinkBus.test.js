import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../redux/store.js';
import { LinkBus } from '../LinkBus.js';

const rowData = {
  original: {
    assigned_bus: null
  }
};

describe('linkBus', () => {
  it('should render Mobile menu', () => {
    const linkBus = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <LinkBus row={rowData} />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(linkBus).toMatchSnapshot();
  });
});
