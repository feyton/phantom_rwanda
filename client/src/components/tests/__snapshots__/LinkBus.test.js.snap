// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`linkBus should render Mobile menu 1`] = `"None"`;
